### ISSUE TITLE
Map 4 or 8 keyboard keys to a vJoy Hat

### ISSUE THREAD

**Issue by E1iTeDa357** [WhiteMagic/JoystickGremlin#74]
I have been trying to find a way to map 4 or 8 keyboard keys to a single vjoy hat but have not figured out an easy way to do it. I also tried my hand at a custom module but no luck. JoystickGremlin is the only solution I have found to work with my 8 axis controller as it does not have the limitations of AHK

**Comment by WhiteMagic** [comment]
With R9 you should be able to do this using macro actions with activation conditions.

For example if you want the right arrow key to correspond to vJoy 1, Hat 1, East you'd do the following:
1. Add a new key input under the Keyboard tab
1. Add two macro actions to this new key input
1. To each of these two actions add a single new vJoy entry for hat 1
1. On one of the actions set it's value to East and the other one to Center
1. Now add conditions, such that the action sending the "East" value is triggered on press and the "Center" one is triggered when releasing the button
Writing this out made me realize that these steps might be hard to follow even though the actual things you want to do are rather simple once you know which bits do what in the UI. I'll make a quick recording and add this in as that, together with the steps above should hopefully be clear enough.

**Comment by E1iTeDa357** [comment]
Thanks!
